app/controllers/activities_controller.rb

```diff
@@ -7,12 +7,31 @@ def index
 
   def create
     @activity = MessageActivity.create(message_params)
-    PusherActivityNotificationJob.perform_later(appointment.guider_id, @activity.id)
     respond_to do |format|
       format.js { render @activity }
     end
   end
 
+  def update_resolve
+    @activity = Activity.find(params[:id])
+    @activity.resolve!(current_user) unless @activity.resolved?
```

Comment:
`resolve!` makes that guard check so we can just remove that postfix modifier and happily return earlier from `resolve!` I reckon.